examples/widgets/text_cursor.py

```diff
@@ -0,0 +1,313 @@
+                # Find closest x value in sorted x vector.
+                # This is the code line,
+                # which requires the plotted data to be sorted.
+                index = np.searchsorted(data, findme)
+                # Return none, if this index is out of range.
+                if (index < 0) or (index >= len(data)):
+                    return None
+                # Return plot point as tuple.
+                return (xdata[index][0], ydata[index][0])
+
+        # Return none if there is no good related point for this x position.
+        return None
+
+    def clear(self, event):
+        """Overwirtten clear callback for cursor. Called right before \
+        displaying the figure."""
+
+        # The base class saves the clean background for blitting.
+        # Text and cursor are invisible,
+        # until the first mouse move event occurs.
+        super().clear(event)
+        if self.ignore(event):
+            return
+        self.text.set_visible(False)
+
+    def _update(self):
+        """
+        Overwritten method for eather blitting or drawing the widget canvas.
+
+        Passes call to base class if blitting is activated, only.
+        In other cases, one draw_idle call is enough, which is placed
+        explicitly in this class (see *onmove()*).
+        In that case, `~matplotlib.widgets.Cursor` is not supposed to draw
+        something using this method.
+        """
+
+        if self.useblit:
+            super()._update()
+
+
+fig = plt.figure(figsize=(8, 6))
+ax = fig.add_subplot(111, facecolor='#FFFFCC')
+
+# A linearly growing x vector.
+x = np.linspace(-5, 5, 1000)
+
+# A non biunique function. dataaxis='y' will cause trouble.
+y = (x ** 2)
+
+lin = ax.plot(x, y)
+ax.set_xlim(-5, 5)
+ax.set_ylim(0, 25)
+
+# A minimum call
+# Set useblit=True on most backends for enhanced performance
+# and pass the ax parameter to the Cursor base class.
+# cursor = TextCursor(line=lin[0], ax=ax, useblit=True)
+
+# A more advanced call. Properties for text and lines are passed.
+# See the color if you are confused which parameter is passed where.
+# The dataaxis parameter is still the default.
+cursor = TextCursor(
+    line=lin[0],
+    numberformat="{0:.2f}\n{1:.2f}",
+    dataaxis='x', offset=[10, 10],
+    textprops={'color': 'blue', 'fontweight': 'bold'},
+    ax=ax,
+    useblit=True,
+    color='red',
+    linewidth=2)
+
+# A call demonstrating problems with the dataaxis=y parameter.
+# The text now looks up the matching x value for the current cursor y position
+# instead of vice versa. Hover you cursor to y=4. There are two x values
+# producing this y value: -2 and 2. The function is only unique,
+# but not biunique. Only one value is shown in the text.
+# cursor = TextCursor(
+#     line=lin[0],
+#     numberformat="{0:.2f}\n{1:.2f}",
+#     dataaxis='y', offset=[10, 10],
+#     textprops={'color': 'blue', 'fontweight': 'bold'},
+#     ax=ax,
+#     useblit=True,
+#     color='red', linewidth=2)
```

Comment:
Not sure if you want to go through this discussion and the added complexity within the example. Alternatively, you could assume that x is the independent variable as is the case for most plots, and let users figure out the other case.

If this stays, this should be moved to a separate paragraph below `plt.show()`. See https://sphinx-gallery.github.io/stable/syntax.html#how-to-structure-your-python-scripts-for-sphinx-gallery

Reply:
I now have this code organized in a second plt.show() and like it's described in the sphinx gallery docs you linked. Can this stay the way I organized it now, or shall I remove the whole hint?